### PR TITLE
Test: direct unit tests for app/services/datadog/client.py

### PR DESCRIPTION
**Pull request by SB2318**
Fixes #878 

 

#### Describe the changes you have made in this PR -

Added a dedicated service-layer test suite for the `DatadogClient` integration to ensure direct coverage of core Datadog API interactions without relying on tool-layer tests.

### Key changes:
- Introduced `tests/services/test_datadog_client.py`
- Added unit tests for:
  - `search_logs()` (success, HTTP failure, empty response)
  - `list_monitors()` (success, HTTP failure, empty response)
  - `get_events()` (success, HTTP failure, empty response)
  - `is_configured` validation
- Mocked `httpx.Client` to ensure no real Datadog API calls are made
- Ensured service-layer isolation so tool tests no longer implicitly validate client logic
- Verified consistent response normalization across logs, monitors, and events APIs

---

### Demo/Screenshot for feature changes and bug fixes - 

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

## Explain your implementation approach:

This PR introduces service-level testing for the Datadog integration to improve test isolation, reliability, and maintainability.

Previously, Datadog client behavior was only indirectly validated through tool-layer tests. This created tight coupling between layers and made failures harder to trace and debug.

---

## What this solves:
- Ensures Datadog API logic is directly tested at the service layer  
- Removes dependency on tool-layer tests for validating core client behavior  
- Improves test clarity, isolation, and maintainability  
- Makes failures easier to debug by localizing responsibility to the service layer  

---

## Implementation approach:
- Used `httpx.Client` mocking to simulate Datadog API responses without making real network calls  
- Covered both success and failure scenarios for each endpoint:
  - logs (`search_logs`)
  - monitors (`list_monitors`)
  - events (`get_events`)  
- Verified response normalization logic (ensuring consistent structure for logs, monitors, and events)  
- Added coverage for configuration state validation via `is_configured`  

---

## Key design choice:

I chose service-layer mocking instead of integration testing to ensure:

- Deterministic and stable test execution in CI  
- No dependency on external Datadog credentials or network availability  
- Faster test execution and improved developer feedback loop  
- Clear separation between service logic and tool-layer orchestration  

This approach ensures that the Datadog client is independently verifiable and does not rely on higher-level tool tests for correctness.

This helps reviewers understand your thought process and ensures you understand the code.
-->

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

File: tests/services/test_datadog_async_client.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    async def post_router(*args, **kwargs):
+        url = str(args[0])
+
+        if "logs" in url:
+            return log_response
+        if "events" in url:
+            return event_response
+
+        raise AssertionError(f"Unexpected POST url: {url}")
```
[the remark this answers is not in the document]
Comment (SB2318) [reply tests/services/test_datadog_async_client.py:66]: haan ji pata hai, apka response structure real issue hai.

**Comment by SB2318** [comment]
@yashksaini-coder, please review.

And, one thing I haven't fixed yet.
image


### Reason

That bot comment is basically informational, not asking for a code change.

It’s saying: you import httpx only to use httpx.HTTPStatusError in tests, which is acceptable, but make sure httpx is available in test dependencies. Since your project already uses httpx as a runtime dependency for the actual client, this is usually already satisfied.

**Comment by muddlebee** [comment]
:eyes: _review in progress_

File: tests/services/test_datadog_client.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    mock_instance = MagicMock()
+
+    mock_response = MagicMock(status_code=500, text="server error")
+
+    mock_instance.post.side_effect = httpx.HTTPStatusError(
+        "error",
+        request=MagicMock(),
+        response=mock_response,
+    )
+
```
Comment (muddlebee) [inline tests/services/test_datadog_client.py:85]: Wrong error injection point. The implementation calls resp.raise_for_status() after post() returns, but here post() itself raises the HTTPStatusError. This accidentally works because the same exception type is caught, but it doesn't exercise the actual code path. Use raise_for_status.side_effect instead, matching the pattern used in test_list_monitors_http_error (line 185).

File: tests/services/test_datadog_client.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+    result = client.search_logs("error")
+
+    assert "logs" in result
+    assert result["success"] is True
+    assert result["logs"][0]["message"] == "log message"
+
+
+def test_search_logs_empty_data(client, mock_httpx_client):
+    mock_instance = MagicMock()
+
```
Comment (muddlebee) [inline tests/services/test_datadog_client.py:57]: test_search_logs_empty_data doesn't assert result["total"] == 0. The implementation always returns total, and the empty path should verify it.

File: tests/services/test_datadog_client.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+# -------------------------
+# list_monitors
+# -------------------------
+
+
+def test_list_monitors_success(client, mock_httpx_client):
+    mock_instance = MagicMock()
+
+    mock_instance.get.return_value = MagicMock(
+        json=lambda: [{"name": "CPU Monitor"}],
```
Comment (muddlebee) [inline tests/services/test_datadog_client.py:117]: mock_instance.post.return_value is dead setup — list_monitors only calls get, never post. Remove it.

File: tests/services/test_datadog_client.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+    mock_instance.get.return_value = MagicMock()
+
+    mock_httpx_client.return_value = mock_instance
+
+    result = client.get_events("error")
+
+    assert result["success"] is True
+    assert result["events"][0]["title"] == "event title"
+
+
```
Comment (muddlebee) [inline tests/services/test_datadog_client.py:200]: mock_instance.get.return_value is dead setup — get_events only calls post, never get. Remove it.

File: tests/services/test_datadog_async_client.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    with patch("app.services.datadog.client.httpx.AsyncClient") as mock:
+        yield mock
+
+
+# -------------------------
+# signature test
+# -------------------------
+
+
+def test_fetch_all_signature_contract():
```
Comment (muddlebee) [inline tests/services/test_datadog_async_client.py]: Signature introspection via inspect.signature tests Python's own reflection, not any client behavior. If the parameter is renamed the test fails but the feature still works; if a parameter is silently added with a wrong type the test still passes. Replace with a behavioral assertion (e.g. verify fetch_all runs end-to-end with expected kwargs) or drop this test.

File: tests/services/test_datadog_async_client.py [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    assert "logs_query" in sig.parameters
+    assert "time_range_minutes" in sig.parameters
+    assert "logs_limit" in sig.parameters
+    assert "monitor_query" in sig.parameters
+    assert "events_query" in sig.parameters
+
+
+# -------------------------
+# validation test
+# -------------------------
```
Comment (muddlebee) [inline tests/services/test_datadog_async_client.py]: This only verifies Python's own enforcement of a required positional argument, not any business logic. Drop it.

**Comment by muddlebee** [comment]
image

**Review by yashksaini-coder — CHANGES_REQUESTED** [review]
## PR #1085: DatadogClient Service-Layer Tests

Ran locally: **25/25 pass**, lint clean, no new typecheck errors (2 pre-existing on `clickhouse.py`/`kafka.py` unrelated to this PR).

---

### ✅ What's Good

- **Greptile P1 — already fixed**: The earlier `result["logs"][0]` `KeyError` in `test_fetch_all_success_strong` is corrected — current line 128 is `result["logs"]["logs"][0]["message"]` which matches the actual `fetch_all` return shape. Good catch and quick fix.
- Coverage is meaningful: success, HTTP error, generic error, empty, and `is_configured` for each of the 3 endpoints.
- `get_pods_on_node` tests are written correctly — they stub `search_logs` directly instead of mocking httpx, which is the right approach for a method that delegates.
- `test_fetch_all_signature_contract` is a nice contract-level guard that catches signature regressions without needing a full mock.
- **Security**: all credentials are clearly fake (`test-api-key`, `test-app-key`). No real network calls anywhere. No hardcoded production secrets.

---

### 🔴 P2 — `test_search_logs_http_error` mocks the wrong layer

**File:** `tests/services/test_datadog_client.py`, lines 75–84

```python
mock_instance.post.side_effect = httpx.HTTPStatusError(
    "error", request=MagicMock(), response=mock_response,
)
```

This fires the error at the `post()` call itself. But the actual client code is:
```python
resp = self._get_client().post(...)  # httpx never raises here
resp.raise_for_status()              # THIS raises on 4xx/5xx
```

In real httpx, `Client.post()` always returns a `Response` object — it is `raise_for_status()` that raises `HTTPStatusError`. The test passes only because `except httpx.HTTPStatusError` catches it anyway, but it exercises a code path that **cannot happen in production**. Compare to how `list_monitors` and `get_events` HTTP error tests do it correctly:

```python
mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(...)
mock_instance.post.return_value = mock_response  # post() succeeds, raise_for_status() raises
```

Please fix `test_search_logs_http_error` to match this pattern:
```python
def test_search_logs_http_error(client, mock_httpx_client):
    mock_instance = MagicMock()
    mock_httpx_client.return_value = mock_instance

    mock_response = MagicMock()
    mock_response.status_code = 500
    mock_response.text = "server error"
    mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(
        "error", request=MagicMock(), response=mock_response,
    )
    mock_instance.post.return_value = mock_response

    result = client.search_logs("error")
    assert result["success"] is False
    assert "HTTP 500" in result["error"]
```

---

### 🟡 P2 — No async `httpx.HTTPStatusError` path tested

`test_fetch_all_partial_failure` uses `Exception("boom")` for all 3 legs. This covers the `except Exception` fallback in each async method — but NOT the `except httpx.HTTPStatusError` blocks, which produce a different error format (`f"HTTP {status}: ..."`). Those branches are never exercised.

Add one async HTTP error test:
```python
@pytest.mark.asyncio
async def test_fetch_all_http_error(async_client, mock_async_httpx):
    mock_instance = MagicMock()
    mock_async_httpx.return_value.__aenter__.return_value = mock_instance

    mock_response = MagicMock()
    mock_response.status_code = 401
    mock_response.text = "unauthorized"
    mock_response.raise_for_status = AsyncMock(
        side_effect=httpx.HTTPStatusError("err", request=MagicMock(), response=mock_response)
    )
    mock_instance.post = AsyncMock(return_value=mock_response)
    mock_instance.get = AsyncMock(return_value=mock_response)

    result = await async_client.fetch_all(
        logs_query="error", time_range_minutes=15,
        logs_limit=100, monitor_query="error", events_query="error",
    )

    assert result["logs"]["success"] is False
    assert "HTTP 401" in result["logs"]["error"]
```

---

### 🟡 Nit — Redundant cross-method mock setup

- `test_list_monitors_success` sets `mock_instance.post.return_value = MagicMock()` — `list_monitors` only calls `get()`, never `post()`. Dead line.
- `test_get_events_success` sets `mock_instance.get.return_value = MagicMock()` — `get_events` only calls `post()`, never `get()`. Dead line.

Remove these — they add noise and suggest copy-paste without checking which HTTP method each endpoint actually uses.

---

### 🟡 Nit — `test_search_logs_empty_data` missing full assertion

```python
result = client.search_logs("error")
assert result["logs"] == []
```

Missing `assert result["success"] is True` and `assert result["total"] == 0`. The other empty-case tests (`test_list_monitors_empty`, `test_get_events_empty`) are complete — be consistent here too.

---

### Summary

| Check | Status |
|---|---|
| 25 tests pass locally | ✅ |
| Lint (ruff) | ✅ Clean |
| Typecheck (mypy) | ✅ No new errors |
| Security (credentials, network) | ✅ No issues |
| Greptile P1 (KeyError in async test) | ✅ Already fixed by contributor |
| `test_search_logs_http_error` mock layer | ❌ Must fix — wrong code path |
| Async `HTTPStatusError` path coverage | ⚠️ Add one test |
| Redundant mock setups | ⚠️ Clean up |
| `test_search_logs_empty_data` completeness | ⚠️ Add 2 assertions |

Fix the `test_search_logs_http_error` mock pattern (the main P2 — it tests an impossible code path) and the two nits, then this is good to merge.

**Comment by SB2318** [comment]
> ## PR #1085: DatadogClient Service-Layer Tests
> Ran locally: **25/25 pass**, lint clean, no new typecheck errors (2 pre-existing on `clickhouse.py`/`kafka.py` unrelated to this PR).
> 
> ### ✅ What's Good
> * **Greptile P1 — already fixed**: The earlier `result["logs"][0]` `KeyError` in `test_fetch_all_success_strong` is corrected — current line 128 is `result["logs"]["logs"][0]["message"]` which matches the actual `fetch_all` return shape. Good catch and quick fix.
> * Coverage is meaningful: success, HTTP error, generic error, empty, and `is_configured` for each of the 3 endpoints.
> * `get_pods_on_node` tests are written correctly — they stub `search_logs` directly instead of mocking httpx, which is the right approach for a method that delegates.
> * `test_fetch_all_signature_contract` is a nice contract-level guard that catches signature regressions without needing a full mock.
> * **Security**: all credentials are clearly fake (`test-api-key`, `test-app-key`). No real network calls anywhere. No hardcoded production secrets.
> 
> ### 🔴 P2 — `test_search_logs_http_error` mocks the wrong layer
> **File:** `tests/services/test_datadog_client.py`, lines 75–84
> 
> ```python
> mock_instance.post.side_effect = httpx.HTTPStatusError(
>     "error", request=MagicMock(), response=mock_response,
> )
> ```
> 
> This fires the error at the `post()` call itself. But the actual client code is:
> 
> ```python
> resp = self._get_client().post(...)  # httpx never raises here
> resp.raise_for_status()              # THIS raises on 4xx/5xx
> ```
> 
> In real httpx, `Client.post()` always returns a `Response` object — it is `raise_for_status()` that raises `HTTPStatusError`. The test passes only because `except httpx.HTTPStatusError` catches it anyway, but it exercises a code path that **cannot happen in production**. Compare to how `list_monitors` and `get_events` HTTP error tests do it correctly:
> 
> ```python
> mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(...)
> mock_instance.post.return_value = mock_response  # post() succeeds, raise_for_status() raises
> ```
> 
> Please fix `test_search_logs_http_error` to match this pattern:
> 
> ```python
> def test_search_logs_http_error(client, mock_httpx_client):
>     mock_instance = MagicMock()
>     mock_httpx_client.return_value = mock_instance
> 
>     mock_response = MagicMock()
>     mock_response.status_code = 500
>     mock_response.text = "server error"
>     mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(
>         "error", request=MagicMock(), response=mock_response,
>     )
>     mock_instance.post.return_value = mock_response
> 
>     result = client.search_logs("error")
>     assert result["success"] is False
>     assert "HTTP 500" in result["error"]
> ```
> 
> ### 🟡 P2 — No async `httpx.HTTPStatusError` path tested
> `test_fetch_all_partial_failure` uses `Exception("boom")` for all 3 legs. This covers the `except Exception` fallback in each async method — but NOT the `except httpx.HTTPStatusError` blocks, which produce a different error format (`f"HTTP {status}: ..."`). Those branches are never exercised.
> 
> Add one async HTTP error test:
> 
> ```python
> @pytest.mark.asyncio
> async def test_fetch_all_http_error(async_client, mock_async_httpx):
>     mock_instance = MagicMock()
>     mock_async_httpx.return_value.__aenter__.return_value = mock_instance
> 
>     mock_response = MagicMock()
>     mock_response.status_code = 401
>     mock_response.text = "unauthorized"
>     mock_response.raise_for_status = AsyncMock(
>         side_effect=httpx.HTTPStatusError("err", request=MagicMock(), response=mock_response)
>     )
>     mock_instance.post = AsyncMock(return_value=mock_response)
>     mock_instance.get = AsyncMock(return_value=mock_response)
> 
>     result = await async_client.fetch_all(
>         logs_query="error", time_range_minutes=15,
>         logs_limit=100, monitor_query="error", events_query="error",
>     )
> 
>     assert result["logs"]["success"] is False
>     assert "HTTP 401" in result["logs"]["error"]
> ```
> 
> ### 🟡 Nit — Redundant cross-method mock setup
> * `test_list_monitors_success` sets `mock_instance.post.return_value = MagicMock()` — `list_monitors` only calls `get()`, never `post()`. Dead line.
> * `test_get_events_success` sets `mock_instance.get.return_value = MagicMock()` — `get_events` only calls `post()`, never `get()`. Dead line.
> 
> Remove these — they add noise and suggest copy-paste without checking which HTTP method each endpoint actually uses.
> 
> ### 🟡 Nit — `test_search_logs_empty_data` missing full assertion
> ```python
> result = client.search_logs("error")
> assert result["logs"] == []
> ```
> 
> Missing `assert result["success"] is True` and `assert result["total"] == 0`. The other empty-case tests (`test_list_monitors_empty`, `test_get_events_empty`) are complete — be consistent here too.
> 
> ### Summary
> Check	Status
> 25 tests pass locally	✅
> Lint (ruff)	✅ Clean
> Typecheck (mypy)	✅ No new errors
> Security (credentials, network)	✅ No issues
> Greptile P1 (KeyError in async test)	✅ Already fixed by contributor
> `test_search_logs_http_error` mock layer	❌ Must fix — wrong code path
> Async `HTTPStatusError` path coverage	⚠️ Add one test
> Redundant mock setups	⚠️ Clean up
> `test_search_logs_empty_data` completeness	⚠️ Add 2 assertions
> Fix the `test_search_logs_http_error` mock pattern (the main P2 — it tests an impossible code path) and the two nits, then this is good to merge.

Okay, I will. Please give me 30 mins.

And please review my This PR also #1086

**Comment by SB2318** [comment]
> ## PR #1085: DatadogClient Service-Layer Tests
> Ran locally: **25/25 pass**, lint clean, no new typecheck errors (2 pre-existing on `clickhouse.py`/`kafka.py` unrelated to this PR).
> 
> ### ✅ What's Good
> * **Greptile P1 — already fixed**: The earlier `result["logs"][0]` `KeyError` in `test_fetch_all_success_strong` is corrected — current line 128 is `result["logs"]["logs"][0]["message"]` which matches the actual `fetch_all` return shape. Good catch and quick fix.
> * Coverage is meaningful: success, HTTP error, generic error, empty, and `is_configured` for each of the 3 endpoints.
> * `get_pods_on_node` tests are written correctly — they stub `search_logs` directly instead of mocking httpx, which is the right approach for a method that delegates.
> * `test_fetch_all_signature_contract` is a nice contract-level guard that catches signature regressions without needing a full mock.
> * **Security**: all credentials are clearly fake (`test-api-key`, `test-app-key`). No real network calls anywhere. No hardcoded production secrets.
> 
> ### 🔴 P2 — `test_search_logs_http_error` mocks the wrong layer
> **File:** `tests/services/test_datadog_client.py`, lines 75–84
> 
> ```python
> mock_instance.post.side_effect = httpx.HTTPStatusError(
>     "error", request=MagicMock(), response=mock_response,
> )
> ```
> 
> This fires the error at the `post()` call itself. But the actual client code is:
> 
> ```python
> resp = self._get_client().post(...)  # httpx never raises here
> resp.raise_for_status()              # THIS raises on 4xx/5xx
> ```
> 
> In real httpx, `Client.post()` always returns a `Response` object — it is `raise_for_status()` that raises `HTTPStatusError`. The test passes only because `except httpx.HTTPStatusError` catches it anyway, but it exercises a code path that **cannot happen in production**. Compare to how `list_monitors` and `get_events` HTTP error tests do it correctly:
> 
> ```python
> mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(...)
> mock_instance.post.return_value = mock_response  # post() succeeds, raise_for_status() raises
> ```
> 
> Please fix `test_search_logs_http_error` to match this pattern:
> 
> ```python
> def test_search_logs_http_error(client, mock_httpx_client):
>     mock_instance = MagicMock()
>     mock_httpx_client.return_value = mock_instance
> 
>     mock_response = MagicMock()
>     mock_response.status_code = 500
>     mock_response.text = "server error"
>     mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(
>         "error", request=MagicMock(), response=mock_response,
>     )
>     mock_instance.post.return_value = mock_response
> 
>     result = client.search_logs("error")
>     assert result["success"] is False
>     assert "HTTP 500" in result["error"]
> ```
> 
> ### 🟡 P2 — No async `httpx.HTTPStatusError` path tested
> `test_fetch_all_partial_failure` uses `Exception("boom")` for all 3 legs. This covers the `except Exception` fallback in each async method — but NOT the `except httpx.HTTPStatusError` blocks, which produce a different error format (`f"HTTP {status}: ..."`). Those branches are never exercised.
> 
> Add one async HTTP error test:
> 
> ```python
> @pytest.mark.asyncio
> async def test_fetch_all_http_error(async_client, mock_async_httpx):
>     mock_instance = MagicMock()
>     mock_async_httpx.return_value.__aenter__.return_value = mock_instance
> 
>     mock_response = MagicMock()
>     mock_response.status_code = 401
>     mock_response.text = "unauthorized"
>     mock_response.raise_for_status = AsyncMock(
>         side_effect=httpx.HTTPStatusError("err", request=MagicMock(), response=mock_response)
>     )
>     mock_instance.post = AsyncMock(return_value=mock_response)
>     mock_instance.get = AsyncMock(return_value=mock_response)
> 
>     result = await async_client.fetch_all(
>         logs_query="error", time_range_minutes=15,
>         logs_limit=100, monitor_query="error", events_query="error",
>     )
> 
>     assert result["logs"]["success"] is False
>     assert "HTTP 401" in result["logs"]["error"]
> ```
> 
> ### 🟡 Nit — Redundant cross-method mock setup
> * `test_list_monitors_success` sets `mock_instance.post.return_value = MagicMock()` — `list_monitors` only calls `get()`, never `post()`. Dead line.
> * `test_get_events_success` sets `mock_instance.get.return_value = MagicMock()` — `get_events` only calls `post()`, never `get()`. Dead line.
> 
> Remove these — they add noise and suggest copy-paste without checking which HTTP method each endpoint actually uses.
> 
> ### 🟡 Nit — `test_search_logs_empty_data` missing full assertion
> ```python
> result = client.search_logs("error")
> assert result["logs"] == []
> ```
> 
> Missing `assert result["success"] is True` and `assert result["total"] == 0`. The other empty-case tests (`test_list_monitors_empty`, `test_get_events_empty`) are complete — be consistent here too.
> 
> ### Summary
> Check	Status
> 25 tests pass locally	✅
> Lint (ruff)	✅ Clean
> Typecheck (mypy)	✅ No new errors
> Security (credentials, network)	✅ No issues
> Greptile P1 (KeyError in async test)	✅ Already fixed by contributor
> `test_search_logs_http_error` mock layer	❌ Must fix — wrong code path
> Async `HTTPStatusError` path coverage	⚠️ Add one test
> Redundant mock setups	⚠️ Clean up
> `test_search_logs_empty_data` completeness	⚠️ Add 2 assertions
> Fix the `test_search_logs_http_error` mock pattern (the main P2 — it tests an impossible code path) and the two nits, then this is good to merge.

Addressed all review points:

* fixed `test_search_logs_http_error` to mock `raise_for_status()` instead of raising from `post()`
* added async `HTTPStatusError` coverage for `fetch_all`
* removed redundant mock setup in monitor/events success tests
* completed assertions for empty search logs case

File: tests/services/test_datadog_client.py (hunk 3 above)
Comment (SB2318) [reply tests/services/test_datadog_client.py:57]: solved

File: tests/services/test_datadog_client.py (hunk 4 above)
Comment (SB2318) [reply tests/services/test_datadog_client.py:117]: solved

File: tests/services/test_datadog_client.py (hunk 5 above)
Comment (SB2318) [reply tests/services/test_datadog_client.py:200]: solved

File: tests/services/test_datadog_client.py (hunk 2 above)
Comment (SB2318) [reply tests/services/test_datadog_client.py:85]: Fixed this in the latest update.

`test_search_logs_http_error` now returns a mocked response from `post()` and raises `httpx.HTTPStatusError` via `raise_for_status.side_effect`, matching the actual production flow and the same pattern used in `test_list_monitors_http_error`.

File: tests/services/test_datadog_async_client.py (hunk 7 above)
Comment (SB2318) [reply tests/services/test_datadog_async_client.py]: Validation test dropped

File: tests/services/test_datadog_async_client.py (hunk 6 above)
Comment (SB2318) [reply tests/services/test_datadog_async_client.py]: Signature test dropped

**Comment by SB2318** [comment]
@yashksaini-coder @muddlebee please review
